src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -1,88 +1,208 @@
+            m_features.shaderSubgroupClustered = subgroupProperties.supportedOperations & VK_SUBGROUP_FEATURE_CLUSTERED_BIT;
+            m_features.shaderSubgroupQuad = subgroupProperties.supportedOperations & VK_SUBGROUP_FEATURE_QUAD_BIT;
+            m_features.shaderSubgroupQuadAllStages = ((subgroupProperties.supportedStages & asset::ISpecializedShader::E_SHADER_STAGE::ESS_ALL)
+                                                        == asset::ISpecializedShader::E_SHADER_STAGE::ESS_ALL);
+        }
+                
         uint32_t qfamCount = 0u;
         vkGetPhysicalDeviceQueueFamilyProperties(m_vkphysdev, &qfamCount, nullptr);
         core::vector<VkQueueFamilyProperties> qfamprops(qfamCount);
         vkGetPhysicalDeviceQueueFamilyProperties(m_vkphysdev, &qfamCount, qfamprops.data());
-
+                
         m_qfamProperties = core::make_refctd_dynamic_array<qfam_props_array_t>(qfamCount);
         for (uint32_t i = 0u; i < qfamCount; ++i)
         {
             const auto& vkqf = qfamprops[i];
             auto& qf = (*m_qfamProperties)[i];
-
+                    
             qf.queueCount = vkqf.queueCount;
             qf.queueFlags = static_cast<E_QUEUE_FLAGS>(vkqf.queueFlags);
             qf.timestampValidBits = vkqf.timestampValidBits;
             qf.minImageTransferGranularity = { vkqf.minImageTransferGranularity.width, vkqf.minImageTransferGranularity.height, vkqf.minImageTransferGranularity.depth };
         }
     }
-
+            
     inline VkPhysicalDevice getInternalObject() const { return m_vkphysdev; }
-
+            
     E_API_TYPE getAPIType() const override { return EAT_VULKAN; }
 
+    std::vector<ISurface::SFormat> getAvailableFormatsForSurface(const ISurface* surface) const override
+    {
+        const ISurfaceVK* vk_surface = static_cast<const ISurfaceVK*>(surface); // Todo(achal): This is problematic, if passed `surface` isn't a vulkan surface
+
+        uint32_t formatCount = 0u;
+        vkGetPhysicalDeviceSurfaceFormatsKHR(m_vkphysdev, vk_surface->m_surface, &formatCount, nullptr);
+        std::vector<VkSurfaceFormatKHR> formats(formatCount);
+        vkGetPhysicalDeviceSurfaceFormatsKHR(m_vkphysdev, vk_surface->m_surface, &formatCount, formats.data());
+
+        std::vector<ISurface::SFormat> result(formatCount);
+
+        for (uint32_t i = 0u; i < formatCount; ++i)
+        {
+            result[i].format = ISurfaceVK::getFormat(formats[i].format);
+            result[i].colorSpace = ISurfaceVK::getColorSpace(formats[i].colorSpace);
+        }
+
+        return result;
+    }
+
+    std::vector<ISurface::E_PRESENT_MODE> getAvailablePresentModesForSurface(const ISurface* surface) const override
+    {
+        const ISurfaceVK* vk_surface = static_cast<const ISurfaceVK*>(surface); // Todo(achal): This is problematic, if passed `surface` isn't a vulkan surface
```

Comment:
@devshgraphicsprogramming did you had a look at this?

Reply:
when an `ISurface` is created it hangs onto a device pointer, get it out of it, and compare it for compatibility here

Reply:
it doesn't

Reply:
it most definitely should